commit 154771ac8bc1f96378aaa093f5ed431932e4bcec
Author: Tor Didriksen <tor.didriksen@oracle.com>
Date:   Thu Apr 27 08:48:15 2023 +0200

    Bug #34278103 Compile MySQL with GCC 13 [noclose]
    
    Downgrade -Werror=suggest-attribute=format to -Wno-error
    
    Change-Id: I087ef84777e136b9bb55dca58365aa1613dc2e06
    (cherry picked from commit 83a5abe7bccb0aeb629e545b50684c7f9acd9444)

diff --git a/storage/ndb/src/common/util/CMakeLists.txt b/storage/ndb/src/common/util/CMakeLists.txt
index b6b9d87cd72..b9b308f9d61 100644
--- a/storage/ndb/src/common/util/CMakeLists.txt
+++ b/storage/ndb/src/common/util/CMakeLists.txt
@@ -84,6 +84,13 @@ FOREACH(tests BaseString Bitmask SparseBitmask Parser HashMap2 LinkedStack
   NDB_ADD_TEST("${tests}-t" "${tests}.cpp" LIBS ndbgeneral)
 ENDFOREACH(tests)
 
+# This looks like a gcc bug, there *are* ATTRIBUTE_FORMAT annotations for
+# cstrbuf::appendf()
+IF(MY_COMPILER_IS_GNU AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 13)
+  TARGET_COMPILE_OPTIONS(cstrbuf-t PRIVATE
+    "-Wno-error=suggest-attribute=format")
+ENDIF()
+
 NDB_ADD_TEST(ndb_version-t version.cpp LIBS ndbgeneral)
 
 FOREACH(tests
